kogito-editors-js/packages/boxed-expression-component/src/api/DataTypeProps.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface DataTypeProps {
+  /** Identifier of the data type, e.g. UNDEFINED */
+  typeRef: string;
+  /** Label used for the data type, e.g. <Undefined> */
+  name: string;
+  /** Tells whether this data type is custom or built-in */
+  isCustom: boolean;
```

Comment:
For all `@jsType` annotated classes, we are using the wrapper classes as field types (e.g. https://github.com/kiegroup/kogito-editors-java/blob/main/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/jsinterop/props/Clause.java#L26) because basically, we have no guarantee by JavaScript that such fields are populated. 
I am not sure how the jsInterop field population is done and I want to avoid blocking errors.